### PR TITLE
created security config, inited Spring boot application, created enti…

### PR DESCRIPTION
**Pull request by VChorney**
…ty user and endpoint with service for registration

### CODE REVIEW

File: src/main/java/com/quizte/QuizteApplication.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.quizte;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class QuizteApplication {
```
Comment (itevas) [inline src/main/java/com/quizte/QuizteApplication.java:7]: maybe QuizterApplication?))

**Comment by itevas** [comment]
Ok.

**Review by itevas — APPROVED** [review]

File: src/main/java/com/quizte/config/QuizteConfig.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import javax.sql.DataSource;
+import java.util.Map;
+
+@Configuration
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class QuizteConfig {
```
Comment (relyativus) [inline src/main/java/com/quizte/config/QuizteConfig.java]: This configuration is redundant. spring-boot-starter-data-jpa will create datasource for you. All you need to do is to set standard properties

File: src/main/java/com/quizte/entity/User.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    @Column
+    String photoUrl;
+
+    @Column(name = "is_active")
+    private Boolean active;
+
+    @CollectionTable(name = "user_role", joinColumns = {@JoinColumn(name = "user_id")})
+    @ElementCollection(fetch = FetchType.EAGER)
+    @Enumerated(EnumType.STRING)
+    Set<Role> roles = new HashSet<>();
```
Comment (relyativus) [inline src/main/java/com/quizte/entity/User.java]: let's keep it simple: user can have only one role. than we can make it a simple column and avoid extra join for roles

File: src/main/java/com/quizte/mapper/UserMapper.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import com.quizte.entity.User;
+import org.modelmapper.ModelMapper;
+
+public class UserMapper {
+    private static final ModelMapper mapper = new ModelMapper();
+
+    private UserMapper() {
+    }
+
+    public static User toUser(RegistrationUserDto registrationUserDto) {
```
Comment (relyativus) [inline src/main/java/com/quizte/mapper/UserMapper.java]: Don't use static methods apart from general utility methods, such as date formatting, string processing, etc. for anything else - use spring components

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,5 @@
+spring.datasource.url=jdbc:postgresql://localhost/Quizte
```
Comment (relyativus) [inline src/main/resources/application.properties:1]: move this to ```application-dev.properties``` so we will be able to configure multiple environments later

**Review by relyativus — CHANGES_REQUESTED** [review]

**Pull request merged**